source/common/conn_pool/conn_pool_base.cc

```diff
@@ -425,18 +425,19 @@ void ConnPoolImplBase::checkForIdleAndCloseIdleConnsIfDraining() {
 
 void ConnPoolImplBase::onConnectionEvent(ActiveClient& client, absl::string_view failure_reason,
                                          Network::ConnectionEvent event) {
-  if (client.state() == ActiveClient::State::CONNECTING) {
-    ASSERT(connecting_stream_capacity_ >= client.effectiveConcurrentStreamLimit());
-    connecting_stream_capacity_ -= client.effectiveConcurrentStreamLimit();
-  }
+  switch (event) {
+  case Network::ConnectionEvent::RemoteClose:
+  case Network::ConnectionEvent::LocalClose: {
+    if (client.state() == ActiveClient::State::CONNECTING) {
+      ASSERT(connecting_stream_capacity_ >= client.effectiveConcurrentStreamLimit());
+      connecting_stream_capacity_ -= client.effectiveConcurrentStreamLimit();
+    }
```

Comment:
Should these changes be reverted along with the other 0-RTT changes?

Reply:
done